### DISCUSSION TITLE
Allow optimizer to run without the budget contstrain

### DISCUSSION THREAD

**Discussion by JalatorrS1** [dcajasn/Riskfolio-Lib#177] [Ideas]
I suggest allowing the budget argument to take the value of None. In this case, the budget constraint will be ignored. I have realized after many tweaks that this constraint becomes a problem when dealing with expanded problems.

**Comment by dcajasn** [comment]
Hi @JalatorrS1,

What kind of problems do you have with the budget constraint?

Best regards,
Dany

**Reply by JalatorrS1** [reply]
If the problem is expanded to include percentage transaction cost, for example. The user is forced to override the portfolio.optimization method. It will be helpful if the budget argument can take the value of None. In this case there is no need to override the method and budget constraints if needed can be included with the  "A" , "b" arguments

**Reply by dcajasn** [reply]
Can you provide an example in a jupyter notebook or python script? The budget constraint is needed because without it portfolios will have a lot of leverage.
